cmd/changelog-extractor/main.go

```diff
@@ -0,0 +1,61 @@
+/*
+   Copyright 2019 Splunk Inc.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+)
+
+func main() {
+	log.SetOutput(os.Stderr)
+	if len(os.Args) != 2 {
+		log.Fatalln("Must pass the Changelog file name as the first argument")
+	}
+	filename := os.Args[1]
+	if err := printReleaseNotes(filename); err != nil {
+		log.Fatalln(err)
+	}
+
+}
+
+func printReleaseNotes(filename string) error {
```

Comment:
if you changed the signature of this function to (a) either accept an `io.Writer` instead of assuming `os.Stdout` or (b) make it return a string containing the changelog for a tag then you would be able to write a real test case for it, rather than just an example.

Reply:
oh i see, you are using the `// Output:` comment to test it. Works for me.

Reply:
Yes, I like Example tests 😁